corehq/apps/reports/views.py

```diff
@@ -1840,7 +1840,10 @@ def _get_form_context(request, domain, instance):
 
 def _get_form_or_404(domain, id):
     try:
-        return FormAccessors(domain).get_form(id)
+        form = FormAccessors(domain).get_form(id)
```

Comment:
@calellowitz having access to forms/cases not within the domain looks pretty insecure.
When using `FormAccessors(domain)` one would expect this accessor to work for only this domain, at least i would. Should we modify `get_form` and other getters to check this internally instead?

May be a soft assert or some email notification to ensure that this is not happening anywhere else?

Reply:
Yeah I think that's a good idea. In addition to the this immediate fix, I'd be in favor of adding a soft assert to within `FormAccessors(domain).get_form(id)` when the resultant form is not in the `domain`, and eventually making that raise the same thing that's raised if the form doesn't exist at all.